core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -181,11 +210,42 @@ public void flush() {
         telemetryBuffer.flush();
     }
 
+    /**
+     * Sets the buffer size
+     * @param maxTelemetriesInBatch should be between MIN_NUMBER_OF_TELEMETRIES_PER_CONTAINER
+     *                              and MAX_NUMBER_OF_TELEMETRIES_PER_CONTAINER inclusive
+     */
+    public void setMaxTelemetriesInBatch(int maxTelemetriesInBatch) {
+        if (maxTelemetriesInBatch >= MIN_NUMBER_OF_TELEMETRIES_PER_CONTAINER &&
```

Comment:
Please consult with Harel regarding the desired behavior here as we might want that if the provided buffer is bigger than the maximum we'll set the value to the maximum value and if it is lower than the minimum then we'll set it to a value lower than the minimum.